### PR TITLE
Introduce fuzzing for `lading_payload::block::Cache::fixed`

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit introduces a new fuzzing harness for the above mentioned function. We want to ensure that we are not crashing at runtime or consuming more memory than allowed.

### Related issues

REF SMPTNG-79

### CODE REVIEW

**Comment by blt** [comment]
Run with `cargo +nightly fuzz run fixed_cache_creation -- -rss_limit_mb=10000 -malloc_limit_mb=5000 -max_len=65535 -jobs=16 -minimize_crash=1`

**Review by GeorgeHahn — APPROVED** [review]

File: lading_payload/src/dogstatsd.rs [hunk 1]
```diff
@@ -147,6 +179,22 @@ impl<T> ConfRange<T>
...
+        match self {
+            Self::Constant(_) => true,
+            Self::Inclusive { min, max } => min < max,
+        }
+    }
+
+    fn start(&self) -> T {
+        match self {
+            ConfRange::Constant(c) => *c,
+            ConfRange::Inclusive { max, .. } => *max,
```
Comment (GeorgeHahn) [inline lading_payload/src/dogstatsd.rs:194]: Should this be `min`?

File: lading_payload/src/block.rs [hunk 2]
```diff
@@ -50,6 +67,26 @@ pub struct Block {
     pub bytes: Bytes,
 }
 
+/// Errors the construction of the block cache
```
Comment (GeorgeHahn) [inline lading_payload/src/block.rs:70]: Word(s) missing

File: lading_payload/fuzz/fuzz_targets/fixed_cache_creation.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    match input.payload {
+        lading_payload::Config::Fluent // slow
+        | lading_payload::Config::OpentelemetryLogs
+        | lading_payload::Config::OpentelemetryTraces
+        | lading_payload::Config::OpentelemetryMetrics
+        | lading_payload::Config::SplunkHec { .. }
+        | lading_payload::Config::Static { .. }
+        | lading_payload::Config::Syslog5424
+        | lading_payload::Config::TraceAgent(..) => return,
+        _ => {}
```
Comment (GeorgeHahn) [inline lading_payload/fuzz/fuzz_targets/fixed_cache_creation.rs:34]: What values pass through this?

File: lading_payload/src/dogstatsd.rs (hunk 1 above)
Comment (blt) [reply lading_payload/src/dogstatsd.rs:194]: D'oh yes.

Comment (blt) [reply lading_payload/src/dogstatsd.rs]: Addressed in 4d4a4a4

File: lading_payload/fuzz/fuzz_targets/fixed_cache_creation.rs (hunk 3 above)
Comment (blt) [reply lading_payload/fuzz/fuzz_targets/fixed_cache_creation.rs]: Addressed in b2f16e0

File: lading_payload/src/block.rs (hunk 2 above)
Comment (blt) [reply lading_payload/src/block.rs]: Addressed in [8ae10f9](https://github.com/DataDog/lading/pull/727/commits/8ae10f9b03d27c7da15813b093cfc36fc63cadb8)

File: lading_payload/src/dogstatsd.rs [hunk 4]
```diff
@@ -219,6 +272,30 @@ pub struct Config {
...
+            && self.tag_value_length.end() > 0
+            && self.name_length.valid()
+            && self.name_length.end() <= MAX_NAME_LENGTH
+            && self.tag_key_length.valid()
+            && self.tag_value_length.valid()
+            && self.tags_per_msg.valid()
+            && self.multivalue_count.valid()
+            && self.value.valid()
+    }
+}
```
Comment (blt) [inline lading_payload/src/dogstatsd.rs:297]: This is the weakest part of this PR. I would like to get these pushed into the deserializer but I'm loath to write a deserializer by hand. Curious for thoughts.

**Review by GeorgeHahn — APPROVED** [review]

**Pull request merged**